test_plans/queue-shortcut-functions.asciidoc

```diff
@@ -0,0 +1,100 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for queue shortcut functions
+
+This is a test plan for the queue shortcut functions as described in SYCL 2020 Section 4.6.5.2.
+
+Estimated development time is four days.
+
+== Testing scope
+
+This plan does not include negative tests.
+
+=== Backend coverage
+
+All the tests described below are performed for any SYCL implementation.
+
+=== Device coverage
+
+All the tests described below are performed only on the default device that is selected on the CTS command line.
+
+== Tests
+Most of the functions tested have three overloads: (1) without dependent events, (2) with a single dependent event, and (3) with multiple dependent events. When this is the case, overload (2) will use the returned event from (1) as its `depEvent` and overload (3) will use the returned events from (1) and (2) as its `depEvents`.
+
+=== Kernel function
+
+The below functions are called twice: with the kernel function as a lambda function and with the kernel function as a named function object type. For each test, the kernel function is some simple kernel that writes a non-zero value to USM. This USM is zero-initialized and is checked with afterwards for correctness. The `parallel_for` tests for `range` are executed for a three-dimensional range of size `1`. The `parallel_for` tests for `nd_range` are executed for a three-dimensional global and local range of size `1`.
```

Comment:
Section 4.6.5.2 of the specification states:
> It is not allowed to capture accessors into the implicitly created command group. If a queue shortcut function launches a kernel (via `single_task` or `parallel_for`), only USM pointers are allowed inside such kernels.

So no, I don't think I can write to memory using an accessor.

Reply:
Good point.  I agree that these tests need to use USM.  I suggest using "device" USM allocations since that is most likely to be supported.  As I note below, the test will need to check if the device supports `aspect::usm_device_allocations`.

Reply:
Alright. I forgot to add this specification for the kernel tests, it is added now.